src/containers/WelcomePage/WelcomePage.jsx

```diff
@@ -163,6 +255,86 @@ export class WelcomePage extends Component {
   }
 }
 
+function mapSearchToFrontPageStructure(data, t) {
+  if (
+    !data ||
+    (data && !data.search) ||
+    (data && data.search && !data.search.results) ||
+    (data &&
+      data.search &&
+      data.search.results &&
+      data.search.results.length === 0)
+  ) {
+    return [];
+  }
+  const result = data.search.results;
+  // grouping
+  const subjects = {
+    title: `${t('contentTypes.subject-material')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  const topics = {
+    title: `${t('subjectPage.tabs.topics')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  const resource = {
+    title: `${t('resource.label')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  result.map(i => {
+    if (i && i.contexts && i.contexts.length) {
+      i.contexts.map(ctx => {
+        if (!ctx.id || !ctx.resourceTypes[0]) {
+          return false;
+        }
+        const finalObj = {
+          id: `${i.id}-${ctx.id}`,
+          path: `/subjects${ctx.path}`,
+          boldName: `${ctx.subject}:`,
+          name: i.title,
+          subName:
+            ctx.resourceTypes[0] && ctx.resourceTypes[0].name
+              ? ctx.resourceTypes[0].name
+              : '',
+        };
+        if (
+          ctx.resourceTypes[0].id === RESOURCE_TYPE_SUBJECT_MATERIAL &&
+          !subjects.resources.filter(obj => obj.path === finalObj.path).length // skip if we already have it
+        ) {
+          subjects.resources.push(finalObj);
+        } else if (
+          ctx.resourceTypes[0].id === RESOURCE_TYPE_LEARNING_PATH &&
+          !topics.resources.filter(obj => obj.path === finalObj.path).length // skip if we already have it
+        ) {
+          topics.resources.push(finalObj);
+        } else if (
+          ctx.resourceTypes[0].id === RESOURCE_TYPE_TASKS_AND_ACTIVITIES &&
+          !resource.resources.filter(obj => obj.path === finalObj.path).length // skip if we already have it
+        ) {
+          resource.resources.push(finalObj);
+        }
+        return false;
+      });
+    }
+    return false;
+  });
+  const returnArray = [];
+  // add groups into return array if there are any resources
+  if (subjects.resources.length) {
```

Comment:
Er ikke så fan av slike falsy måter å sjekke ting på. Vet at 0 tolkes som falsy, men for lesbarheten sin skyld, tenker jeg at det spesifikt sjekkes på lengde === 0.